src/coreclr/jit/importercalls.cpp

```diff
@@ -130,6 +130,47 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
         // assume the worst-case.
         mflags = (calliSig.callConv & CORINFO_CALLCONV_HASTHIS) ? 0 : CORINFO_FLG_STATIC;
 
+        if (opts.OptimizationEnabled() && call->AsCall()->unmgdCallConv == CorInfoCallConvExtension::Managed &&
+            ((GenTree*)call->AsCall()->gtCallMethHnd)->OperIs(GT_FTN_ADDR))
```

Comment:
```suggestion
        if (opts.OptimizationEnabled() && !call->AsCall()->IsUnmanaged() &&
            call->AsCall()->gtCallAddr->OperIs(GT_FTN_ADDR))
```

Reply:
I don't think we know at this stage whether the call will be a pinvoke (`impCheckForPInvokeCall` happens after this), so the `IsUnmanaged()` check will always be false.

Reply:
It turns out that the `IsUnmanaged` check is actually unnecessary. If the callconv (static vs non-static), args type and return type are all match and the calling target is a managed method, we can just ignore the unmanaged calling conv as both caller and callee are managed.